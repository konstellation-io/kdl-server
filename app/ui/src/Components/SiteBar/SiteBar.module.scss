@import 'Styles/variables';
@import 'Styles/mixins';
@import 'Styles/colors';

.container {
  display: flex;
  height: 5 * $grid-unit;
  background-color: palette(base, 900);
  white-space: nowrap;

  z-index: 999;
}

.left,
.right {
  display: flex;
  z-index: 3;
}

.serverDetails {
  position: relative;
  display: flex;
  flex-grow: 1;
  align-items: center;
  align-self: center;
  padding-left: 0.5 * $grid-unit;
  height: 100%;

  .serverName {
    display: flex;
    align-items: center;
    border-right: solid 2px palette(base, 500);

    .serverLabel {
      @include font-caption;
      color: font-color(regular);

      align-self: center;
      margin-right: $grid-unit;
    }

    .serverState {
      @include make-triangle(2 * $grid-unit, palette(success));

      margin-right: $grid-unit;
    }
    .serverNameInput {
      @include font-subheader;
      color: font-color(light);
    }
  }
}
