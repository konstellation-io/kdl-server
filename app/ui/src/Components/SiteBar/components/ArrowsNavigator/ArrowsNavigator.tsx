import * as React from 'react';
import styles from './ArrowsNavigator.module.scss';
import ArrowBack from '@material-ui/icons/ArrowBackIos';
import ArrowForward from '@material-ui/icons/ArrowForwardIos';
import { Button } from 'kwc';
import { useHistory } from 'react-router-dom';

function ArrowsNavigator() {
  const { goBack, goForward } = useHistory();

  return (
    <div className={styles.container}>
      <Button label="" Icon={ArrowBack} onClick={goBack} />
      <Button label="" Icon={ArrowForward} onClick={goForward} />
    </div>
  );
}

export default ArrowsNavigator;
