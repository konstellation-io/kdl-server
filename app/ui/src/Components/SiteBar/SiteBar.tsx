import { Left, Right } from 'kwc';
import * as React from 'react';

import ArrowsNavigator from './components/ArrowsNavigator/ArrowsNavigator';
import Breadcrumbs from './components/Breadcrumbs/Breadcrumbs';
import SettingsMenu from '../SettingsMenu/SettingsMenu';
import styles from './SiteBar.module.scss';
import isElectron from 'is-electron';
import RuntimesCrumb from './components/Breadcrumbs/components/RuntimesCrumbs/RuntimesCrumb';
import CapabilitiesCrumb from './components/Breadcrumbs/components/CapabilitiesCrumbs/CapabilitiesCrumb';

const SiteBar = () => (
  <div className={styles.container}>
    <Left className={styles.left}>
      <>
        {isElectron() && <ArrowsNavigator />}
        <Breadcrumbs />
      </>
    </Left>
    <Right className={styles.right}>
      <RuntimesCrumb title={'Runtimes'} />
      <CapabilitiesCrumb title={'Capabilities'} />
      <SettingsMenu />
    </Right>
  </div>
);

export default SiteBar;
