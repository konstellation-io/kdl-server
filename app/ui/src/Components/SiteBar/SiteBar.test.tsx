import * as React from 'react';
import ArrowsNavigator from './components/ArrowsNavigator/ArrowsNavigator';
import SiteBar from './SiteBar';
import { shallow } from 'enzyme';
import isElectron from 'is-electron';

jest.mock('is-electron', () => jest.fn(() => false));

let component = shallow(<SiteBar />);

describe('SiteBar component', () => {
  test('Component match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should hide navigation buttons when running on browser', function () {
    expect(component.find(ArrowsNavigator).isEmptyRender()).toBeTruthy();
  });

  it('should show navigation buttons when running on electron', function () {
    // Act.
    isElectron.mockReturnValue(true);
    component = shallow(<SiteBar />);

    // Assert.
    expect(component.find(ArrowsNavigator).isEmptyRender()).toBeFalsy();
  });
});
