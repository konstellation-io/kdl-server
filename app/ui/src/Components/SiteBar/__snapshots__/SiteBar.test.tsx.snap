// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteBar component Component match snapshot 1`] = `
<div
  className="container"
>
  <ot
    className="left"
  >
    <Breadcrumbs />
  </ot>
  <ht
    className="right"
  >
    <RuntimesCrumb
      title="Runtimes"
    />
    <CapabilitiesCrumb
      title="Capabilities"
    />
    <Memo(SettingsMenu) />
  </ht>
</div>
`;
